### DISCUSSION TITLE
Subfolder routing in /static

### DISCUSSION THREAD

**Discussion by justindasilva** [nuxt/nuxt#16931] [General]
This is a Nuxt3 question.

I have a build process that generates a small app (It's a SlideV presentation) and throws it into my main app in the `static` folder so that when I build and deploy my main app, I can see the Presentation at `/presentation`

Everything is working pretty well.

However, if I try visiting `/presentation/4` for example (trying to visit the 4th slide in the Presentation), the app will load up in stead of my presentation and show a 404 error.

Is there some way for me to define (maybe in the `nuxt.config.ts`?) that I want all routes with `/presentation/*` to use the static app at `/presentation`?

**Comment by danielroe** [comment]
In Nuxt 3, the folder has been renamed `public/`. You can mark any links to it as external or use a middleware to force a hard reload.

**Comment by justindasilva** [comment]
Sorry, I develop using Nuxt 2 and Nuxt 3 and I mistyped. I throw the Presentation app into the `public` folder.

And also, I should fix what I said above, because I can see how you thought I was asking about getting to `/presentation/4` through my app. I'll re-explain below.

What if I pasted the following URL in my browser: `http://localhost:3000/presentation/4`.
When I do this currently, my main application tries to render the route, but in fact, I want my `/public/presentation` which is a generated app to instead render the route.

**Reply by Lyokolux** [reply]
I am in the same situation with a UI component documentation ([Histoire](https://histoire.dev/)). The app is accessed at `/docs/stories`. However, accessing the direct URL `/docs/stories/my-beautiful-story-example` returns a 404 error.

What we need: a way to delegate the `/my-beautiful-story-example` routing part to the static app.

**Reply by akasummer** [reply]
That sounds more like a problem with your hosting server setup. What is happening is that your server is trying to locate my-beautiful-story-example.html file in stories/ which is missing, since you have a single page application. To resolve these routing issues, you need to configure your server to rewrite all requests to /index.html file (I suppose your built app contains only this one). How to do that rewrite will depend on the hosting provider you use. Just google "SPA create rewrite rule" and the name of your web server.

**Reply by justindasilva** [reply]
These next couples of screenshots might tell a bit of a different story. (By the way, I've uploaded all my code to https://github.com/justindasilva/nuxt-plus-docs where you can see this example)

I click on a link from my app that goes to `/presentation`. I can then navigate to the second slide within this separate static app which brings me to `/presentation/2` (this works fine)
![Screen Shot 2022-09-06 at 3 10 58 PM](https://user-images.githubusercontent.com/882534/188719465-a3d53c55-da2a-41d9-b9e6-6648c109e7f8.png)

If I go to `http://localhost:3001/presentation/2` in my browser, Nuxt 3 is now handling this URL as seen below:
![Screen Shot 2022-09-06 at 3 10 40 PM](https://user-images.githubusercontent.com/882534/188719769-af5e8649-b0e8-46be-9a67-83cd3d1e0ec4.png)

Is there no way in the Nuxt Config to do something like this?

```
import { defineNuxtConfig } from 'nuxt'

export default defineNuxtConfig({
   
   extendRoutes: [
      {
          path: '/presentation/*',
          handledBy: '/presentation',
     }
   ]

})
```

This basically says, user going to any URL that starts with `/presentation/` should be handled by the `presentation` app which exists in `/public/presentation`

**Comment by meadowsys** [comment]
I believe [nuxt server routes](https://v3.nuxtjs.org/guide/features/server-routes) may be able to achieve this? You can try creating a route in the file `server/routes/presentation/[[...slug]].ts`, and in the handler read and return the index.html file of the presentation app. I believe in your presentation app you have set the base path to `/presentation` so that the assets link up properly, but if not, you can set `app.baseURL` in `nuxt.config.ts` to `/presentation/`.

Explanation for why I did `[[...slug]].ts` with the double square brackets and spread slug thing (to give my reasoning in case it doesn't work (I have not tested)):
- Docs say that the spread operator in `[...slug].vue` matches all routes under that path, including nested ones
- Double square brackets make the parameter optional, so `pages/test/[[slug]].vue` will match `/test/nested` as well as `/test`

edit: I just realised, my answer is assuming that presentation is a fully client-side rendered SPA. The concepts I presented here would probably still apply and be useful if it were SSR, just would need more finessing with
